Update compiler.md (#43872)
vercel/next.js: docs/advanced-features/compiler.md

### Jest

> Jest not only includes the transformation previously provided by Babel, but also simplifies configuring Jest together with Next.js including:

Reviewer: ```suggestion
The Next.js Compiler also performs transformation for your Jest tests and simplifies configuring Jest together with Next.js including:
```
